src/SprykerEco/Zed/Braintree/Business/Payment/Transaction/AbstractTransaction.php

```diff
@@ -57,7 +57,7 @@ public function executeTransaction(TransactionMetaTransfer $transactionMetaTrans
      */
     protected function getTransactionIdentifier()
     {
-        return $this->transactionMetaTransfer->requireTransactionIdentifier()->getTransactionIdentifier();
+        return $this->transactionMetaTransfer->requireTransactionIdentifier()->getTransactionIdentifierOrFail();
```

Comment:
this line is wrong
`require` is mixed with `orFail`

Reply:
I thought that there could be a possible BC break since using `require` instead of `OrFail` would change the class of an exception thrown here in case of an empty transaction identifier. 

But as you explained to me in a private slack discussion, changing an exception class is not a BC break. So I fixed this by using `OrFail` without `require`.